spring-integration-redis/src/main/java/org/springframework/integration/redis/outbound/RedisStreamOutboundChannelAdapter.java

```diff
@@ -0,0 +1,58 @@
+package org.springframework.integration.redis.outbound;
+
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.connection.stream.ObjectRecord;
+import org.springframework.data.redis.connection.stream.StreamRecords;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.data.redis.core.StreamOperations;
+import org.springframework.data.redis.serializer.JdkSerializationRedisSerializer;
+import org.springframework.data.redis.serializer.RedisSerializer;
+import org.springframework.data.redis.serializer.StringRedisSerializer;
+import org.springframework.integration.handler.AbstractMessageHandler;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+/**
+ * Take a message from a Channel and publish it to the given Redis Stream
+ */
+public class RedisStreamOutboundChannelAdapter extends AbstractMessageHandler {
+
+    private final RedisTemplate<String, Object> template;
+
+    private final StreamOperations streamOperations;
+
+    private final String                        streamName;
+
+    private volatile RedisSerializer<?> serializer = new JdkSerializationRedisSerializer();
+
+    public RedisStreamOutboundChannelAdapter( String streamName, RedisConnectionFactory connectionFactory ) {
+        Assert.notNull( streamName, "'streamName' must not be null" );
+        Assert.notNull( connectionFactory, "'connectionFactory' must not be null" );
+        this.template = new RedisTemplate<>();
+        this.template.setConnectionFactory( connectionFactory );
+        this.template.setKeySerializer( new StringRedisSerializer() );
+        this.template.setValueSerializer( this.serializer );
+        template.afterPropertiesSet();
+        this.streamOperations = template.opsForStream();
+        this.streamName = streamName;
+    }
+
+    public void setSerializer( RedisSerializer<?> serializer ) {
+        Assert.notNull( serializer, "'serializer' must not be null" );
+        this.serializer = serializer;
+    }
+
+    @Override
+    public String getComponentType() {
+        return "redis:stream-outbound-channel-adapter";
+    }
+
+    @Override
+    protected void handleMessageInternal( Message<?> message ) {
+        ObjectRecord<String, Message<?>> record = StreamRecords
+                .<String, Message<?>> objectBacked( message )
+                .withStreamKey( streamName );
+        this.streamOperations.add( record );
```

Comment:
@artembilan How can I improve this code?
Currently I don't make any difference about incoming data type. I also publish all the message as is because I want to keep its headers.
Also I have some troubles about how well converting and reading data. 
Say I have that class:
```
public class SimpleUser implements Serializable {
    private final String nom;
    private final String prenom;
    private final String username;
}
```
When publishing a message with a paylod of type `SimpleUser ` I am unable to get it from the inbound part (see my comment above). In the inbound side, I would like to extract it before sending it to the output channel.